apluslms_roman/builder.py

```diff
@@ -32,11 +34,35 @@ def get_steps(self, refs: list = None):
                 raise ValueError(_("A step index was too big. Remember, indexing begins with 0."))
         return list(OrderedDict.fromkeys(steps))
 
-    def build(self, step_refs: list = None):
+    def build(self, step_refs: list = None, host_path=None):
         backend = self._engine.backend
         observer = self._observer
         steps = self.get_steps(step_refs)
-        task = BuildTask(self.path, steps)
+        # Check if has global config set.
+            # Using path in global config file
+        print("dir mapping set in config:", self._engine._dir_mapping if hasattr(self._engine, '_dir_mapping') else None)
+        print("-p paramater set:", host_path)
+        if self._engine._environment.environ.get('DOCKER_DIRECTORY_MAP', None) is not None:
+            if hasattr(self._engine, '_dir_mapping'):
+                print("using config")
+            else:
+                print("Using environment variable")
+            mapping = self._engine._environment.environ.get('DOCKER_DIRECTORY_MAP', None)
+            path = get_host_path(self.path, dict(mapping))
```

Comment:
This should be about the only thing that is needed here, I presume.

Reply:
You mean this should not be changed?

Reply:
I presume when you do the rest of the stuff, the code around this comment will get pretty clean and this seemed to be the only important part.

Though, the dict copy in `dict(mapping)` seems unnecessary.

Reply:
Actually, there is no need to add anything to this file abouf `directory_map` spicifically.

You could add

```
def remap_path(self, path):
  map_ = self.environment.get('directory_map', {})
  return get_host_path(path, map_)
```

To Backend in https://github.com/apluslms/roman/blob/072f95e2bafee900e3c5789bb9fc5fa653cabb53/apluslms_roman/backends/__init__.py#L84